src/rpc.rs

```diff
@@ -573,8 +606,105 @@ impl RPC {
         }
     }
 
-    pub fn notify(&self) {
-        self.notification.send(Notification::Periodic).unwrap();
+    /// Attempt to notify scripthash subscriptions affected by transaction.
+    /// The `notified` hashset is an optimization to avoid double notifications.
```

Comment:
Do you think it would be easier if instead of passing an accumulated `HashSet` of notified script hashes, this function would return a set of script hashes to the caller, and the _caller_ would be the one iterating over it and sending the notifications?

Reply:
Maybe? :-)

Reply:
Ok, I get that :)